components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,342 @@
+              .x(d => this.xScale( d.daysAgo) )
+              .y(d => this.yScale( d.percentage) );
+  }
+
+  get viewBox() {
+    return `0 0 ${this.width} ${this.height}`;
+  }
+
+    ////////////////// RENDER FUNCTIONS ////////////////////
+  private renderChart() {
+    this.AllUnlockDeactivate();
+    this.resizeChart();
+    this.renderGrid();
+    this.renderLine();
+    this.renderPoints();
+    this.renderTooltips();
+    this.renderRings();
+    this.renderLabelButtons();
+    this.relock();
+  }
+
+  private renderLine(): void {
+    // create the line using path function
+    const line = this.createPath(this.data);
+
+    const theLine = this.svgSelection.selectAll('.line').data([this.data], d => d.daysAgo);
+    theLine.exit().remove();
+    theLine.enter().append('path').attr('class', 'line').merge(theLine)
+      .transition().duration(1000)
+    .attr('d', line);
+  }
+
+  private renderPoints(): void {
+    const points = this.svgSelection.selectAll('circle.point')
+      .data(this.data, d => d.daysAgo);
+    points.exit().remove();
+    points.enter().append('circle')
+        .attr('class', (_d, i) => `point elem-${i}`)
+        .merge(points)
+        .transition().duration(1000)
+        .attr('percent', ( d => d.percentage ) )
+        .attr('cx', d => this.xScale(d.daysAgo))
+        .attr('cy', d => this.yScale(d.percentage))
+        .attr('r', 4);
+  }
+
+  private renderRings(): void {
+    const rings = this.svgSelection.selectAll('circle.ring')
+      .data(this.data, d => d.daysAgo);
+    rings.exit().remove();
+    rings.enter().append('circle')
+      .attr('class', (_d, i) => `ring elem-${i}`)
+      .merge(rings)
+      .attr('cx', d => this.xScale(d.daysAgo))
+      .attr('cy', d => this.yScale(d.percentage))
+      .attr('r', 10);
+  }
+
+  private renderTooltips(): void {
+    // these numbers are specific to its container
+    const localWidth = this.width - this.margin.right - this.margin.left - 34;
+    const thisRange = [localWidth, -this.margin.right];
+    const thisScale = d3.scaleLinear()
+      .domain(this.domainX)
+      .range(thisRange);
+
+    const tooltips = this.containerSelection.selectAll('div.graph-tooltip')
+      .data(this.data, d => d.daysAgo);
+    tooltips.exit().remove();
+    tooltips.enter().append('div')
+      .attr('class', (_d, i: number) => `graph-tooltip elem-${i}`)
+      .merge(tooltips)
+      .text(d => `Checked in ${Math.round(d.percentage)}%`)
+      .style('left', (d, i: number) => {
+        const left = thisScale(d.daysAgo);
+        if ( i === 0 ) { return `${localWidth + this.margin.right}px`; }
+        return `${left}px`;
+      })
+      .style('top', d => `${this.yScale(d.percentage) + this.margin.top}px`);
+  }
+
+  private renderLabelButtons(): void {
+    // these numbers are specific to its container
+    const thisRange = [this.width - 48, 53];
+    const thisScale = d3.scaleLinear()
+      .domain(this.domainX)
+      .range(thisRange);
```

Comment:
I applied the same trick for the buttons for the same reason, however the given UX styles required this anyways.